docs/docs/what-is-resourceFactory.md

```diff
@@ -0,0 +1,123 @@
+# What is `resourceFactory`?
+
+## Why do we need it?
+
+`resourceFactory` provides a compatibility with the design goal of resources given the current limitations of public APIs in Ember. The idea is that we'd eventually get rid of it, and even right now, it's basically a no-op.
+
+The behavior that we _want_ when we define a resource like this:
+```js 
+function LocalizedClock(locale) {
+    return resource(() => {
+        /* ... */
+        return 'theValue';
+    });
+}
+```
+and invoke it like this:
+```hbs 
+{{LocalizedClock 'en-US'}}
+```
+
+We want, not only the function, `LocalizedClock` to be invoked, but also the `resource`.
+
+One way to think about this is to strip away all the resource-isms and you're left with:
+```js 
+function localizedClock(locale) {
+    return function myInnerFunction() {
+        return 'theValue';
+    }
+}
+```
+this would still be invoked via:
+```hbs 
+{{localizedClock 'en-US'}}
+```
+but we can more easily see that the return value is a function.
+
+So, like with `LocalizedClock`, we have to do this for `localizedClock`:
+```hbs
+{{#let (localizedClock 'en-US') as |innerFunction|}}
+
+    {{ (innerFunction) }} -- prints theValue
```

Comment:
Maybe add a note about how we are invoking the function inside the curlies, since it's not quite clear why this is needed initially (I didn't quite follow the necessity of the parens inside until the next example).

Reply:
Thanks for the feedback! However, why wasn't it clear? :thinking: I usually consider the spaces between `{{` and `(` to add clarity better than the `prettier` defaults.

it's just the syntax of the template is kind of out of scope for these docs, and is covered by: 

https://tutorial.glimdown.com/1-introduction/3-transforming-data

Reply:
I was just initially wondering why the parens were needed, and assumed that `innerFunction` doesn't execute when used as `{{innerFunction}}`.

Reply:
right, and we want the function to be invoked so `theValue` prints